### DISCUSSION TITLE
analytical hessian

### DISCUSSION THREAD

**Discussion by lucasem1** [mir-group/nequip#292] [Ideas]
Dear Developers

is there a Nequip branch that can calculate the hessian analytically? 

Best regards.
Luca Sementa

**Comment by Linux-cpp-lisp** [comment]
Hi Luca,

This can be achieved by adapting a copy of `PartialForceOutput` (https://github.com/mir-group/nequip/blob/main/nequip/nn/_grad_output.py#L160) to use `torch.autograd.functional.hessian` instead of `torch.autograd.functional.jacobian`. Please note that due to limitations of the underlying PyTorch implementations, this can be quite slow depending on your system and model. Please also note that for the same reason TorchScript deployment when `torch.autograd.functional.*` is used is **not** supported.

If you write a `HessianOutput` class based on `PartialForceOutput`, please feel free to submit a PR against the `develop` branch after reading `CONTRIBUTING.md`; I'd be very happy to merge such an implementation into `nequip`.

Thanks!
